src/main/java/com/commercetools/sync/sdk2/producttypes/utils/AttributeDefinitionUpdateActionUtils.java

```diff
@@ -0,0 +1,374 @@
+                .attributeName(oldAttributeDefinition.getName())
+                .newValue(inputHint)
+                .build());
+  }
+
+  /**
+   * Compares the attribute constraints of an {@link AttributeDefinition} and an {@link
+   * AttributeDefinitionDraft} and returns an {@link
+   * com.commercetools.api.models.product_type.ProductTypeUpdateAction} as a result in an {@link
+   * java.util.Optional}. If both the {@link AttributeDefinition} and the {@link
+   * AttributeDefinitionDraft} have the same attribute constraints, then no update action is needed
+   * and hence an empty {@link java.util.Optional} is returned.
+   *
+   * <p>Note: Only following changes are supported: SameForAll to None and Unique to None. See the
+   * docs:
+   * https://docs.commercetools.com/api/projects/productTypes#change-attributedefinition-attributeconstraint
+   *
+   * @param oldAttributeDefinition the attribute definition which should be updated.
+   * @param newAttributeDefinition the attribute definition draft where we get the new attribute
+   *     constraint.
+   * @return A filled optional with the update action or an empty optional if the attribute
+   *     constraints are identical.
+   * @throws BuildUpdateActionException If newAttributeDefinition has other values than {@code
+   *     AttributeConstraintEnum.NONE}
+   */
+  @Nonnull
+  static Optional<ProductTypeUpdateAction> buildChangeAttributeConstraintUpdateAction(
+      @Nonnull final AttributeDefinition oldAttributeDefinition,
+      @Nonnull final AttributeDefinitionDraft newAttributeDefinition)
+      throws UnsupportedOperationException {
+
+    final AttributeConstraintEnum newAttributeConstraint =
+        newAttributeDefinition.getAttributeConstraint();
+    final AttributeConstraintEnum oldAttributeConstraint =
+        oldAttributeDefinition.getAttributeConstraint();
+    if (newAttributeConstraint != null
+        && !newAttributeConstraint.equals(oldAttributeConstraint)
+        && !AttributeConstraintEnum.NONE.equals(newAttributeConstraint)) {
+      throw new UnsupportedOperationException(
+          format(
+              "Invalid AttributeConstraint update to %s. Only following updates are allowed: SameForAll to None and Unique to None.",
+              newAttributeConstraint.name()));
+    }
+
+    if (AttributeConstraintEnum.NONE.equals(oldAttributeConstraint)
+        && newAttributeConstraint == null) {
+      // attribute constraint is optional and by default attribute constraint is NONE:
+      // https://docs.commercetools.com/api/projects/productTypes#ctp:api:type:AttributeDefinitionDraft
+      return Optional.empty();
+    }
+
+    return buildUpdateAction(
+        oldAttributeConstraint,
+        newAttributeConstraint,
+        () ->
+            ProductTypeChangeAttributeConstraintActionBuilder.of()
+                .attributeName(oldAttributeDefinition.getName())
+                .newValue(AttributeConstraintEnumDraft.NONE)
```

Comment:
I don't want to put the logic of the platform validation here. The only reason I have this check here is because for `changeAttributeConstraint` action I have to supply `AttributeConstraintEnumDraft` class and this class contains only 1 enum - `AttributeConstraintEnumDraft.NONE`. 

**Longer explanation:**
When customer supplies drafts to us, they supply with class `AttributeConstraintEnum`. This class contains 4 enums - NONE, UNIQUE, COMBINATION_UNIQUE and SAME_FOR_ALL. However, when we generate `changeAttributeConstraint` inside java-sync, we have to supply `AttributeConstraintEnumDraft.NONE` only as there is no other enums available for `AttributeConstraintEnumDraft`. Therefore I have to check in this method that user does NOT supply anything else than `AttributeConstraintEnum.NONE`. But the goal was NOT to simulate the platform check here.
If customer supplies `oldAttributeConstraint eq CombinationUnique -> change to NONE` - we would generate an action and it's up to the platform to deal with it.

Reply:
got it. thanks for the explanation.